test/PackageVersionsTest/InstallerTest.php

```diff
@@ -93,6 +93,122 @@ public function testGetSubscribedEvents() : void
         }
     }
 
+    public function testDumpVersionsClassIfExistingFileIsNotWritable() : void
+    {
+        $config            = $this->getMockBuilder(Config::class)->disableOriginalConstructor()->getMock();
+        $locker            = $this->getMockBuilder(Locker::class)->disableOriginalConstructor()->getMock();
+        $repositoryManager = $this->getMockBuilder(RepositoryManager::class)->disableOriginalConstructor()->getMock();
+        $installManager    = $this->getMockBuilder(InstallationManager::class)->disableOriginalConstructor()->getMock();
+        $repository        = $this->createMock(InstalledRepositoryInterface::class);
+
+        $vendorDir = sys_get_temp_dir() . '/' . uniqid('InstallerTest', true);
+
+        $expectedPath = $vendorDir . '/ocramius/package-versions/src/PackageVersions';
+
+        /** @noinspection MkdirRaceConditionInspection */
+        mkdir($expectedPath, 0777, true);
+
+        $expectedFileName = $expectedPath . '/Versions.php';
+        touch($expectedFileName);
+        chmod($expectedFileName, 0444);
+
+        $locker
+            ->expects(self::any())
+            ->method('getLockData')
+            ->willReturn([
+                'packages' => [
+                    [
```

Comment:
Try to simplify the data provider: this test focuses on atomic file writes, so reduce everything that isn't strictly necessary to the minimum please